sdk/search/Azure.Search.Documents/src/Batching/InitializationBarrierSlim.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Search.Documents.Batching
+{
+    /// <summary>
+    /// InitializationBarrierSlim allows for synchronized lazy initialization.  An
```

Comment:
I used to do it and then started looking at our ref docs with links to themselves...  I'm on the fence because I guess those links work in other places so maybe it's still worth going?

Reply:
It may look weird to link to yourself, but if you ever rename classes/members in VS, It Just Works. Also keep in mind that once upon a time, members got their own page, so linking back to the class actually went to the ToC for all members. Our doc formats have changed so many times, I imagine someday we'll go back to that when we realize that classes with many members aren't mobile-/bandwidth-friendly.

Reply:
That said, it's a nit/idea. I don't mind either way.

Reply:
It's a good point.  I'll get back in the habit.